app/src/androidTest/java/com/google/samples/apps/topeka/activity/SignInActivityTest.java

```diff
@@ -65,15 +71,33 @@ public void checkFab_initiallyNotDisplayed() {
     }
 
     @Test
+    public void signIn_checkIfEditTextViewsAreInitiallyEmpty() {
+        onView(withId(R.id.first_name)).check(matches(withText(isEmptyOrNullString())));
+        onView(withId(R.id.last_initial)).check(matches(withText(isEmptyOrNullString())));
+    }
+
+    @Test
     public void signIn_performSuccessful() {
         inputData();
-        onView(withId(R.id.done)).check(matches(isDisplayed()));
+        onView(withId(R.id.done)).check(matches(isDisplayed())).perform(click());
+        onView(withId(R.id.toolbar_player)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar() {
+        for (int i = 0; i < 16; i++) {
+            onData(anything())
+                    .inAdapterView(withId(R.id.avatars))
+                    .atPosition(15)
+                    .check(matches(isDisplayed()))
```

Comment:
@keyboardsurfer about magic numbers, I changed this:
https://github.com/googlesamples/android-topeka/search?utf8=%E2%9C%93&q=magic

```
    private static Quiz createQuizDueToType(Cursor cursor) {
        final String type = cursor.getString(cursor.getColumnIndex(QuizTable.COLUMN_TYPE));
        final String question = cursor.getString(cursor.getColumnIndex(QuizTable.COLUMN_QUESTION));
        final String answer = cursor.getString(cursor.getColumnIndex(QuizTable.COLUMN_ANSWER));
        final String options = cursor.getString(cursor.getColumnIndex(QuizTable.COLUMN_OPTIONS));
        final int min = cursor.getInt(cursor.getColumnIndex(QuizTable.COLUMN_MIN));
        final int max = cursor.getInt(cursor.getColumnIndex(QuizTable.COLUMN_MAX));
        final int step = cursor.getInt(cursor.getColumnIndex(QuizTable.COLUMN_STEP));
        final boolean solved = getBooleanFromDatabase(
                cursor.getString(cursor.getColumnIndex(QuizTable.COLUMN_SOLVED)));
```